### PR TITLE
[US1.5] water-level sensor added

### PR DESCRIPTION
**Pull request by ijkeleher**
Fixed up issues from previous pull request.

Water sensor set up to detect water level.

run ```$ python output.py``` to see data flow in stdout.

### CODE REVIEW

**Comment by ijkeleher** [comment]
lol, so yeah this pi-default user is gonna be our guy until i learn to properly rebase

**Review by auspham — COMMENTED** [review]
Should we push `correct_author.sh` since it's not related to the project ?

**Comment by ijkeleher** [comment]
hmm i would say so just in case something happens to the pi and we have to clone the repo. the script is part of the "infrastructure" in a way.

**Comment by ijkeleher** [comment]
but happy to wait actually. and just rebase this properly. because it looks a bit messy.

**Comment by auspham** [comment]
It looks like it is just correcting the author rather than storing credentials to clone. However keep it since might be useful for development.

> hmm i would say so just in case something happens to the pi and we have to clone the repo. the script is part of the "infrastructure" in a way.

I'll let @nugyn to decide.

> but happy to wait actually. and just rebase this properly. because it looks a bit messy.

**Comment by ijkeleher** [comment]
@rockmanvnx6 @nugyn 

I cleaned it up properly. See new commit history.

So we have two options for developing our code for the arduino sensors as I see it. Maybe more...

1. SSH into the pi which is connected to the arduino and do development on a local branch and then create an upstream branch which we use to make a pull request. This may represent a problem with using multiple users concurrently logged in via ssh I'm not sure. I also don't like this approach because any development is done "live" and since an arduino can only run one program at a time then this basically stops it being used until the development is done. Only useful at a very early stage or if we are going to develop within certain defined "no service" windows.

2. Do development on local machines and use an arduino simulator, then we merge to dev branch and at a certain schedule e.g. every monday we push all changes from dev to prod. I can set up a deployment pipeline so that any push to production branch will activate a script to compile and deploy the new code onto the real arduino which is connected to a pi at my house:

```your machine -> git push --set-upstream origin feature_name -> pull req -> approved to dev -> on monday dev is pushed to prod -> deploy to arduino via pi serial connection```

This is a video of a free arduino emulator that can be used.
https://www.youtube.com/watch?v=UlcvqkYCsvE

What's everyone's preference?

**Comment by auspham** [comment]
maybe development on `2` but testing on `1`

File: water_sensor/water_sensor.ino
Comment (auspham) [inline water_sensor/water_sensor.ino]: indentation

File: water_sensor/water_sensor.ino
Comment (auspham) [inline water_sensor/water_sensor.ino:5]: remove extra line

**Comment by ijkeleher** [comment]
@rockmanvnx6 i think testing should be done in the dev branch, so when a pull request like this is made then all tests will be run automatically and any build fails or test failures will block the pull request. if we mess up our code and push to the arduino then the damage will have been done to the production environment which will damage the integrity of the real data we are gathering.

**Review by nugyn — APPROVED** [review]

**Comment by auspham** [comment]
Please format the code before merging.

**Comment by ijkeleher** [comment]
@rockmanvnx6 take a look at main.ino

**Comment by nugyn** [comment]
Is the PR Request still be modfied. I am okay to merge it.

**Comment by ijkeleher** [comment]
merge please.

**Review by auspham — APPROVED** [review]

File: water_sensor/water_sensor.ino
Comment (ijkeleher) [reply water_sensor/water_sensor.ino:5]: outdated, doesn't matter.

**Pull request merged**
